android/src/main/java/com/applovin/reactnative/AppLovinMAXNativeAdViewManager.java

```diff
@@ -0,0 +1,148 @@
+package com.applovin.reactnative;
+
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReadableArray;
+import com.facebook.react.bridge.ReadableMap;
+import com.facebook.react.common.MapBuilder;
+import com.facebook.react.uimanager.ThemedReactContext;
+import com.facebook.react.uimanager.ViewGroupManager;
+import com.facebook.react.uimanager.annotations.ReactProp;
+
+import java.util.Map;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+public class AppLovinMAXNativeAdViewManager
+        extends ViewGroupManager<AppLovinMAXNativeAdView>
+{
+    public static final String REACT_CLASS = "AppLovinMAXNativeAdView";
+
+    public final int COMMAND_LOAD_AD = 1;
+
+    public AppLovinMAXNativeAdViewManager(final ReactApplicationContext callerContext) { }
+
+    @NonNull
+    @Override
+    protected AppLovinMAXNativeAdView createViewInstance(@NonNull final ThemedReactContext reactContext)
+    {
+        return new AppLovinMAXNativeAdView( reactContext );
+    }
+
+    @NonNull
+    @Override
+    public String getName()
+    {
+        return REACT_CLASS;
+    }
+
+    /// Callback to JavaScript
+
+    @Nullable
+    @Override
+    public Map<String, Object> getExportedCustomDirectEventTypeConstants()
+    {
+        return MapBuilder.<String, Object>builder()
+                .put( "onAdLoaded", MapBuilder.of( "registrationName", "onAdLoaded" ) )
+                .build();
+    }
+
+    /// Call from JavaScript
+
+    @Nullable
+    @Override
+    public Map<String, Integer> getCommandsMap()
+    {
+        return MapBuilder.of(
+                "loadAd", COMMAND_LOAD_AD
+        );
+    }
+
+    // NOTE: the method is deprecated but the new version won't work
+    @Override
+    public void receiveCommand(@NonNull final AppLovinMAXNativeAdView root, final int commandId, @Nullable final ReadableArray args)
+    {
+        switch ( commandId )
+        {
+            case COMMAND_LOAD_AD:
+                root.loadAd();
+                break;
+        }
+    }
+
+    /// Property Setters
+
+    @ReactProp(name = "adUnitId")
+    public void setAdUnitId(final AppLovinMAXNativeAdView view, final String value)
+    {
+        view.setAdUnitId( value );
+    }
+
+    @ReactProp(name = "placement")
+    public void setPlacement(final AppLovinMAXNativeAdView view, @Nullable final String value)
+    {
+        view.setPlacement( value );
+    }
+
+    @ReactProp(name = "customData")
+    public void setCustomData(final AppLovinMAXNativeAdView view, @Nullable final String value)
+    {
+        view.setCustomData( value );
+    }
+
+    @ReactProp(name = "extraParameter")
```

Comment:
Should be extra parameters right?

Reply:
Thanks.  this has to be "extraParameters".   I have verified "extraParameters" on both Android and iOS.